build_tools/cmake/iree_hal_cts_test_suite.cmake

```diff
@@ -30,18 +39,92 @@ function(iree_hal_cts_test_suite)
   cmake_parse_arguments(
     _RULE
     ""
-    "DRIVER_NAME;DRIVER_REGISTRATION_HDR;DRIVER_REGISTRATION_FN"
+    "DRIVER_NAME;DRIVER_REGISTRATION_HDR;DRIVER_REGISTRATION_FN;COMPILER_TARGET_BACKEND;EXECUTABLE_FORMAT"
     "DEPS;EXCLUDED_TESTS;LABELS"
     ${ARGN}
   )
 
+  # Omit tests for which the specified driver is not enabled.
+  string(TOUPPER ${_RULE_DRIVER_NAME} _UPPERCASE_DRIVER)
+  string(REPLACE "-" "_" _NORMALIZED_DRIVER ${_UPPERCASE_DRIVER})
+  if(NOT DEFINED IREE_HAL_DRIVER_${_NORMALIZED_DRIVER})
+    message(SEND_ERROR "Unknown driver '${_RULE_DRIVER_NAME}'. Check IREE_HAL_DRIVER_* options.")
+  endif()
+  if(NOT IREE_HAL_DRIVER_${_NORMALIZED_DRIVER})
+    return()
+  endif()
+
   list(APPEND _RULE_LABELS "driver=${_RULE_DRIVER_NAME}")
 
+  # Enable executable tests if a compiler target backend capable of producing
+  # executables compatible with the driver is provided and enabled.
+  set(_ENABLE_EXECUTABLE_TESTS OFF)
+  if(DEFINED _RULE_COMPILER_TARGET_BACKEND)
+    string(TOUPPER ${_RULE_COMPILER_TARGET_BACKEND} _UPPERCASE_TARGET_BACKEND)
+    string(REPLACE "-" "_" _NORMALIZED_TARGET_BACKEND ${_UPPERCASE_TARGET_BACKEND})
+    if(NOT DEFINED IREE_TARGET_BACKEND_${_NORMALIZED_TARGET_BACKEND})
+      message(SEND_ERROR "Unknown backend '${_RULE_COMPILER_TARGET_BACKEND}'. Check IREE_TARGET_BACKEND_* options.")
+    endif()
+    if(DEFINED IREE_HOST_BINARY_ROOT)
```

Comment:
This could be a single OR condition. Are you just breaking it out for the comments? Not sure whether the extra control flow complexity is worth it, but maybe. Mostly making sure that was a deliberate choice.

Reply:
Good point. I forked this from https://github.com/google/iree/blob/f066e650472d78e506317e505d21fe1e7f6d9619/build_tools/cmake/iree_check_test.cmake#L223-L248

could collapse that condition, but I do like the consistency...